addon/validators/base.js

```diff
@@ -45,28 +45,37 @@ export default Ember.Object.extend({
   isValid: Ember.computed.empty('errors.[]'),
   isInvalid: Ember.computed.not('isValid'),
   validate: function() {
-    var self = this;
-    return this._validate().then(function(success) {
-      // Convert validation failures to rejects.
-      var errors = get(self, 'model.errors');
-      if (success) {
-        return errors;
-      } else {
-        return Ember.RSVP.reject(errors);
-      }
+    return this._validate().then((success) => {
+      const errors = this.errors;
+      return success ? errors : Ember.RSVP.reject(errors);
```

Comment:
We don't use anything from the `Ember` object, it needs to be destructured. Also, destructure from `RSVP`